### PR TITLE
[ Bug] Memory leak fixes on AudioSessionManager

### PR DESCRIPTION
**Pull request by Mohtasim**
## Purpose

* Fixes the memory leak of audio session manager

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test

* Check with android profiler to see memory usage

## What to Check
Verify that the following are valid
* Memory hold while the bluetooth is turned off

### CODE REVIEW

**Review by ahammer — DISMISSED** [review]

**Review by ahammer — COMMENTED** [review]
Is issue multiple registrations of profile proxy? I think in onCreate hook we can check savedInstanceState == null to only register on initial launch, and in isFinishing block to release?

File: azure-communication-ui/azure-communication-ui/src/calling/java/com/azure/android/communication/ui/calling/presentation/manager/AudioSessionManager.kt [hunk 1]
```diff
@@ -38,6 +38,7 @@ internal class AudioSessionManager(
     private val audioManager by lazy { context.getSystemService(Context.AUDIO_SERVICE) as AudioManager }
     private var bluetoothAudioProxy: BluetoothHeadset? = null
     private var initialized = false
+    private var isProfileProxyOpened = false
 
     private val isBluetoothScoAvailable
         get() = try {
```
Comment (phileo99) [inline azure-communication-ui/azure-communication-ui/src/calling/java/com/azure/android/communication/ui/calling/presentation/manager/AudioSessionManager.kt:44]: can you add enclosing brackets in line46 to fix Android studio warning

Comment (phileo99) [inline azure-communication-ui/azure-communication-ui/src/calling/java/com/azure/android/communication/ui/calling/presentation/manager/AudioSessionManager.kt:44]: can you add enclosing brackets in line46 to fix Android studio warning

**Review by phileo99 — CHANGES_REQUESTED** [review]

File: azure-communication-ui/azure-communication-ui/src/calling/java/com/azure/android/communication/ui/calling/presentation/manager/AudioSessionManager.kt [hunk 2]
```diff
@@ -112,26 +112,37 @@ internal class AudioSessionManager(
...
+                BluetoothAdapter.ACTION_CONNECTION_STATE_CHANGED -> openProfileProxy()
                 AudioManager.ACTION_HEADSET_PLUG -> updateHeadphoneStatus()
             }
         }
     }
 
+    private fun openProfileProxy() {
+        if (!isProfileProxyOpened && btAdapter?.state == BluetoothAdapter.STATE_ON) {
+            btAdapter?.run {
+                getProfileProxy(context, this@AudioSessionManager, BluetoothProfile.HEADSET)
```
Comment (phileo99) [inline azure-communication-ui/azure-communication-ui/src/calling/java/com/azure/android/communication/ui/calling/presentation/manager/AudioSessionManager.kt:140]: what should happen if `getProfileProxy()` returns false?

**Review by phileo99 — CHANGES_REQUESTED** [review]

File: azure-communication-ui/azure-communication-ui/src/calling/java/com/azure/android/communication/ui/calling/presentation/manager/AudioSessionManager.kt (hunk 2 above)
Comment (phileo99) [reply azure-communication-ui/azure-communication-ui/src/calling/java/com/azure/android/communication/ui/calling/presentation/manager/AudioSessionManager.kt:140]: ok, I looked at the SDK source code for `getProfileProxy()`, ignore my above comment, it should actually be returning a reference to the BT Headset object, but it doesn't.

File: azure-communication-ui/azure-communication-ui/src/calling/java/com/azure/android/communication/ui/calling/presentation/manager/AudioSessionManager.kt (hunk 1 above)
Comment (Mohtasim) [reply azure-communication-ui/azure-communication-ui/src/calling/java/com/azure/android/communication/ui/calling/presentation/manager/AudioSessionManager.kt:44]: Do you mean {} ? 
There's an enclosing one on line 47.

**Review by iaulakh — CHANGES_REQUESTED** [review]
Hi @Mohtasim,

I noticed below issues that was not reproduced on develop/beta:

1) for multiple bluetooth deices disconnecting one, second does not shows connected
2) for single bluetooth device, disconnecting during call does not change icon to audio or speaker

**Review by ahammer — DISMISSED** [review]
Worked in testing

File: azure-communication-ui/azure-communication-ui/src/calling/java/com/azure/android/communication/ui/calling/presentation/manager/AudioSessionManager.kt [hunk 3]
```diff
@@ -22,6 +22,7 @@ import kotlinx.coroutines.flow.collect
 import android.media.AudioDeviceInfo
 import android.os.Build
 import android.os.Bundle
+import android.util.Log
```
Comment (iaulakh) [inline azure-communication-ui/azure-communication-ui/src/calling/java/com/azure/android/communication/ui/calling/presentation/manager/AudioSessionManager.kt:25]: can be removed

**Review by ahammer — DISMISSED** [review]

File: azure-communication-ui/azure-communication-ui/src/calling/java/com/azure/android/communication/ui/calling/presentation/manager/AudioSessionManager.kt (hunk 3 above)
Comment (Mohtasim) [reply azure-communication-ui/azure-communication-ui/src/calling/java/com/azure/android/communication/ui/calling/presentation/manager/AudioSessionManager.kt]: Thanks :)

File: azure-communication-ui/azure-communication-ui/src/calling/java/com/azure/android/communication/ui/calling/presentation/manager/AudioSessionManager.kt (hunk 1 above)
Comment (Mohtasim) [reply azure-communication-ui/azure-communication-ui/src/calling/java/com/azure/android/communication/ui/calling/presentation/manager/AudioSessionManager.kt:42]: changed as per discussion.

**Review by ahammer — APPROVED** [review]

**Review by iaulakh — APPROVED** [review]

File: azure-communication-ui/azure-communication-ui/src/calling/java/com/azure/android/communication/ui/calling/presentation/manager/AudioSessionManager.kt [hunk 4]
```diff
@@ -286,12 +286,16 @@ internal class AudioSessionManager(
         }
     }
 
+    private fun openProfileProxy() {
+        if (btAdapter!!.isEnabled)btAdapter?.run {
```
Comment (phileo99) [inline azure-communication-ui/azure-communication-ui/src/calling/java/com/azure/android/communication/ui/calling/presentation/manager/AudioSessionManager.kt:290]: this works, but can we clean it up like so:
```
btAdapter?.run { 
      if (isEnabled) getProfileProxy()
}
 ```

**Review by phileo99 — DISMISSED** [review]

**Pull request merged**